sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/ComponentIndexer.java

```diff
@@ -56,7 +58,11 @@ public void execute(DefaultModuleFileSystem fs) {
       boolean unitTest = InputFile.Type.TEST == inputFile.type();
       Resource sonarFile = File.create(inputFile.relativePath(), languages.get(languageKey), unitTest);
       sonarIndex.index(sonarFile);
-      resourceCache.get(sonarFile).setInputPath(inputFile);
+      BatchComponent file = componentCache.get(sonarFile);
+      file.setInputPath(inputFile);
+      Resource sonarDir = file.parent().resource();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'parent' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)